src/main/java/com/streamr/client/dataunion/DataUnion.java

```diff
@@ -0,0 +1,171 @@
+package com.streamr.client.dataunion;
+
+import com.streamr.client.dataunion.contracts.DataUnionMainnet;
+import com.streamr.client.dataunion.contracts.DataUnionSidechain;
+import com.streamr.client.dataunion.contracts.IERC20;
+import com.streamr.client.utils.Web3jUtils;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.web3j.abi.TypeEncoder;
+import org.web3j.abi.datatypes.*;
+import org.web3j.abi.datatypes.generated.Uint256;
+import org.web3j.crypto.Credentials;
+import org.web3j.crypto.ECKeyPair;
+import org.web3j.crypto.Sign;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.tx.Contract;
+import org.web3j.tx.ManagedTransaction;
+import org.web3j.utils.Numeric;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.math.BigInteger;
+import java.util.ArrayList;
+
+import static com.streamr.client.utils.Web3jUtils.*;
+
+public class DataUnion {
+    private static final Logger log = LoggerFactory.getLogger(DataUnion.class);
+    protected static final long STATUS_NONE = 0;
+    protected static final long STATUS_ACTIVE = 1;
+    protected static final long STATUS_INACTIVE = 2;
+
+    // Contract.web3j is protected, so need to access with reflection. more bad web3j design!
+    private static Web3j getWeb3j(Contract c) {
+        try {
+            return (Web3j) FieldUtils.readField(c, "web3j", true);
+        } catch (Exception e) {
+            log.error(e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    private DataUnionMainnet mainnet;
+    private DataUnionSidechain sidechain;
+
+    //use DataUnionClient to instantiate
+    protected DataUnion(DataUnionMainnet mainnet, DataUnionSidechain sidechain) {
+        this.mainnet = mainnet;
+        this.sidechain = sidechain;
+    }
+
+    public boolean waitForDeployment(long sleeptime, long timeout) throws Exception {
```

Comment:
From these two given parameters, I understood that this method would
1) first wait for `sleeptime` (doing nothing, just idle)
2) then do something once a while (not specified how often)
3) finally stop observing the deployment after `sleeptime`+`timeout` or `timeout` (not clear which)

Found out from `waitForCondition` method that my interpretation is wrong.

Suggestion: rename the `sleeptime` to '`pollInterval`' as we are actually doing polling here.